### PR TITLE
Remove mandatory password cb

### PR DESCRIPTION
**Pull request by julek-wolfssl**
The password callback was being used on every decode attempt. First try to decode without pass then ask for it.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -5319,13 +5318,22 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
...
         #endif
+        }
+    #endif /* HAVE_ED25519 */
+
+        if (ret != 0 || (!rsaKey && !eccKey && !ed25519Key)) {
+#if defined(WOLFSSL_ENCRYPTED_KEYS) && !defined(NO_PWDBASED)
+            if (!usePasswd && format != WOLFSSL_FILETYPE_PEM && info->passwd_cb && algId == 0) {
+                usePasswd = 1;
+                WOLFSSL_MSG("Retrying decode with password callback");
+                goto retry_decode;
```
Comment (SparkiDev) [inline src/ssl.c]: Avoid goto like this if possible.

**Comment by dgarske** [comment]
I am pulling this PR into the OpenVPN PR https://github.com/wolfSSL/wolfssl/pull/2503. Sean, also plan to eliminate the goto. Once I push in 2503 will close this PR.

**Comment by julek-wolfssl** [comment]
I removed the goto and moved part of the code to a separate function. I 
hope this solution is better.

**Comment by dgarske** [comment]
Pulled into #2503 for openvpn port.

**Pull request closed without merging**